### ISSUE TITLE
Converting our documentation theme to the pandas_sphinx_theme

### ISSUE THREAD

**Issue by choldgraf** [jupyterhub/team-compass#225]
We've now had several of our documentation sites converted over to the Pandas sphinx theme. My personal feeling is that the new layout is cleaner and more well-organized. If other folks agree, shall we use this issue to track converting the rest of our docs over? I think it'd be good to have a unified documentation theme for our repositories.

- [x] https://jupyterhub.readthedocs.io/en/latest
- [ ] https://jupyterhub-kubespawner.readthedocs.io/en/latest/?badge=latest#
- [x] https://z2jh.jupyter.org
- [x] https://tljh.jupyter.org
- [x] https://binderhub.readthedocs.io/en/latest/
- [x] https://repo2docker.readthedocs.io/en/latest
- [x] https://mybinder.readthedocs.io/en/latest
- [x] https://jupyterhub-team-compass.readthedocs.io
- [x] https://mybinder-sre.readthedocs.io/en/latest/

## How to change the theme

In case others would like to help with this, changing the theme is basically the following:

1. Add `git+https://github.com/pandas-dev/pandas-sphinx-theme.git@master` to the requirements installed for documentation
2. Set the `html_theme` configuration value to `pandas_sphinx_theme` (e.g., https://github.com/jupyterhub/team-compass/blob/master/docs/conf.py#L92)
3. Remove old theme-specific options (these aren't supported in the new theme)
4. Organize our pages so that they have clear levels to them. The main `index.rst` page should have `toctrees` for all top-level sections. For each page that `index.rst` points to, in *that* page, there should be another `toctree` with second-level pages. 
  
  For example, [here is the team compass index page](https://github.com/jupyterhub/team-compass/blob/master/docs/index.rst), and [here's the index page for the team governance 2nd-level section](https://github.com/jupyterhub/team-compass/blob/master/docs/index-team_governance.rst)

**Comment by betatim** [comment]
I like the new layout so moving to it gets my support. A thing to remember is that if we move pages around we should install a redirect so that external links continue to work. In https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/1472#issuecomment-552760608 and follow up comments we figured out how to do this in RTD.

**Comment by choldgraf** [comment]
Good point @betatim - another note here is that the redirects will only happen if we decide to move files into sub-folders, though I think that the Z2JH repository was particularly egregious in its use of a totally flat folder hierarchy so this may not be as much of a problem in the other repositories

**Comment by yuvipanda** [comment]
I too like the new theme. Thanks for working on it, @choldgraf.

I started working on moving this for TLJH in https://github.com/jupyterhub/the-littlest-jupyterhub/pull/468. However, I'm not quite sure what to do about the top level headings. TLJH has a bunch of narrative docs in its index...

**Comment by manics** [comment]
The left-hand panel looks empty on the home page of the new docs- in particular it's missing a search box. Other pages look OK.

[JupyterHub Stable](https://jupyterhub.readthedocs.io/en/stable/):
![image](https://user-images.githubusercontent.com/1644105/70735176-94665d80-1d05-11ea-9673-48b4b13aaee6.png)

[JupyterHub Latest](https://jupyterhub.readthedocs.io/en/latest/):
![image](https://user-images.githubusercontent.com/1644105/70735202-a3e5a680-1d05-11ea-8ab4-3a28ed9ecfd8.png)

**Comment by sgibson91** [comment]
@choldgraf already has a PR addressing the search bar issue open https://github.com/pandas-dev/pandas-sphinx-theme/pull/55 Hopefully the pandas team merge it soon!

**Comment by manics** [comment]
Edit on GitHub button: https://github.com/pandas-dev/pydata-bootstrap-sphinx-theme/issues/62

(Linking so we can see all related issues in this issue).

**Comment by choldgraf** [comment]
@manics that one should now be fixed actually :-)

**Comment by manics** [comment]
Just Kubespawner left in the checklist.

**Comment by choldgraf** [comment]
I think we can consider this one done - if somebody wants to finish up that last repo on top then go for it but I don't think it's worth tracking this anymore
